test/driver-specs.js

```diff
@@ -14,3 +17,35 @@ describe('driver', () => {
   });
 });
 
+describe('getDeviceTime', withMocks({fs, teen_process}, (mocks) => {
+    it('should return a valid date string', async () => {
+      let udid = 'some-udid';
+      let date = new Date().toString();
+      mocks.fs.expects('which')
+        .once().returns('/path/to/idevicedate');
+      mocks.teen_process.expects('exec')
+        .once().withExactArgs('idevicedate', ['-u', udid])
+        .returns({stdout: date});
+      let driver = new IosDriver();
+      driver.opts = {udid};
+
+      (await driver.getDeviceTime()).should.equal(date);
+    });
+    it('should not return a valid date string', async () => {
+      let udid = 'some-udid';
+      let date = new Date().toString();
+      mocks.fs.expects('which')
+        .once().returns('/path/to/idevicedate');
```

Comment:
This test ought to do the case of no `libimobiledevice` installed. That would mean that `exec` would throw an error saying that it couldn't find `idevicedate`.

You can mock this like:

```js
mocks.teen_process.expects("exec")
  .once()
  .throws("ENOENT");
let driver = new IosDriver();
driver.opts = {udid};
await adb.getDeviceTime()
  .should.eventually.be.rejectedWith("Could not capture device date and time");
```